File: apps/external-apis/aldi-api/src/domain/entities/Product.ts
export type Product = {
    id: string;
    title: string;
    description: string;
    category: string;
    price: string; //just to be different from continent-api which price is float
    image: string;
    currency: string;
    brand: string;
    inStock: boolean;
    createdAt: Date;
    updatedAt: Date;
}

File: apps/external-apis/lidl-api/src/Lidl.Domain/Entities/Product.cs
namespace Lidl.Domain.Entities;

public sealed class Product
{
    public Guid Id { get; init;}
    public string Name { get; init; }
    public string Description { get; init; }
    public string Category { get; init; }
    public decimal Price { get; init; }
    public string Currency { get; init; }
    public bool InStock { get; init; }
    public string? Brand { get; init; }
    public string? Image { get; init; }
    public DateTime CreatedAt { get; init; }
    public DateTime UpdatedAt { get; init; }
}